acceptance/tests/windows/stop_puppet_svc_stops_puppet.rb

```diff
@@ -0,0 +1,48 @@
+test_name '(C14941) Stopping puppet service on Windows stops puppet' do
+
+  confine :to, :platform => 'windows'
+  
+  opts = {
+    :acceptable_exit_codes => [0],
+  }
+  
+  agents.each do
+    step 'Verify the puppet service is running' do
+      opts[:acceptable_exit_codes] << 32
+      start = on(agent, 'sc query puppet', opts)
+      unless start.stdout.include?('RUNNING')
+        repeat_fibonacci_style_for 5 do
+          start = on(agent, 'sc start puppet', opts)
+          break if start.stdout.include?('RUNNING')
+        end
+      end
+      query = on(agent, 'sc query puppet', opts)
+      assert_match(/RUNNING/, query.stdout, 'Failed to start service')
+    end
+   
+    step 'Stop the puppet service and verify it stopped' do
+      stop = on(agent, 'sc stop puppet', opts)
+      unless stop.stdout.include?('STOPPED')
+        repeat_fibonacci_style_for 5 do
+          stop = on(agent, 'sc query puppet', opts)
+          break if stop.stdout.include?('STOPPED')
+        end
+      end
```

Comment:
this block is repeated across the three steps.  could we use a local method of `def wait_for_windows_puppet_run_state(state='STOPPED')` or similar?
which would be a good example of possible reuse when someone finds it...